src/transformation/visitors/binary-expression/destructuring-assignments.ts

```diff
@@ -36,28 +38,31 @@ export function isArrayLength(
 export function transformDestructuringAssignment(
     context: TransformationContext,
     node: ts.DestructuringAssignment,
-    root: lua.Expression
+    root: lua.Expression,
+    rootHasPrecedingStatements: boolean
 ): lua.Statement[] {
-    return transformAssignmentPattern(context, node.left, root);
+    return transformAssignmentPattern(context, node.left, root, rootHasPrecedingStatements);
 }
 
 export function transformAssignmentPattern(
     context: TransformationContext,
     node: ts.AssignmentPattern,
-    root: lua.Expression
+    root: lua.Expression,
+    rootHasPrecedingStatements: boolean
```

Comment:
Why does this have to be a separate argument, couldnt we just inspect the context to see if this is the case?